### PR TITLE
Fixes part of #5352: Blog redesign next steps

### LINKED ISSUE #5352: Blog redesign next steps and discussion
We're looking to improve the blog template again with some design inspiration from this WordPress template:

image

We'd like to keep it really minimal but want to:

* add a sidebar including recent posts
* show only the first 300px of the post content, with a `Read more` button that toggles the rest to be visible
* include social media links in sidebar
* include a list of recent tweets in sidebar

We'd love help chopping this up into pieces to implement! This is a pretty simple set of changes that would make some good first-timers-only issues!

### PR DESCRIPTION
**Pull request by Cadreia**
Fixes #5352 (<=== Add issue number here)
blog-plots2

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by Cadreia** [comment]
@publiclab/reviewers, please can you review this pull request?

**Comment by Cadreia** [comment]
@cesswairimu can you please review this pr?

**Comment by jywarren** [comment]
Hiiii @Cadreia --- i'm so sorry we missed this! This looks really great. I'm reviewing it -- will go through more carefully tomorrow!

**Comment by jywarren** [comment]
I just resolved the conflicts to get this merge-able. Then i'll try it out!

File: app/controllers/tag_controller.rb [hunk 1]
```diff
@@ -251,6 +251,15 @@ def widget
     render layout: false
   end
 
+  def blog2
+    nids = Tag.find_nodes_by_type(params[:id], 'note', nil).collect(&:nid)
+    @pagy, @notes = pagy(Node.where('status = 1 AND nid in (?)', nids).order('created DESC'), items: 6)
+    @tags = Tag.where(name: params[:id])
+    @tagnames = @tags.collect(&:name).uniq! || []
+    @title = @tagnames.join(',') + ' Blog' if @tagnames
+    @alltags = Tag.first(6)
```
Comment (jywarren) [inline app/controllers/tag_controller.rb:260]: This looks really great - i'm just going to make a small suggested change here so we can set the featured topics manually. Thanks, @Cadreia !!!!

```suggestion
```

File: app/views/comments/_form.html.erb [hunk 2]
```diff
@@ -36,96 +36,97 @@
...
-                <a onClick="handleClick()" style="cursor: pointer;" class="d-none d-md-inline text-underline">choose one</a>
-                <span class="d-md-none">
-                  <i class="fa fa-upload"></i>
-                  <a><%= translation('comments._form.upload_image') %></a>
-                </span>
-              </label>
-            </span>
-          </p>
-
+                <% if is_new_contributor %>
```
Comment (jywarren) [inline app/views/comments/_form.html.erb]: I think maybe this section was an unintended change?

File: app/views/tag/blog2.html.erb [hunk 3]
```diff
@@ -1,150 +1,182 @@
...
-      <b>Click here to subscribe!</a>
+<div style="width: 100%; height: auto">
+   <div style="text-align:center;">
+      <div class="blog2-header" style="margin-top: 10px;">
+         <p>Public lab logo (size=65x65)</p>
+         <h3><b>The Public Lab Blog</b></h3>
+         <hr>
+         <p>
+            <a href="/wiki/blog">ABOUT</a>
+            <a href="/research">RESEARCH</a>
```
Comment (jywarren) [inline app/views/tag/blog2.html.erb:9]:
```suggestion
            <a href="/tags">TOPICS</a>
```

File: app/views/tag/blog2.html.erb [hunk 4]
```diff
@@ -1,150 +1,182 @@
...
-              <ul>
-              <li>Dummy Text 1 </li>
-              <li>Dummy Text 2 </li>
-              <li>Dummy Text 3 </li>
-              <li>Dummy Text 4 </li>
-              <li>Dummy Text 5 </li>
-              <li>Dummy Text 6 </li>
-              </ul>
+            <div class="col-12 pt-4">
+               <div id="categories-content">
```
Comment (jywarren) [inline app/views/tag/blog2.html.erb:117]: I'll make a direct edit to this section...

**Review by jywarren — CHANGES_REQUESTED** [review]
Just making some smaller tweaks, and replacing the Categories with an admin-modifiable Feature code snippet! This looks great, though!!!!

**Review by jywarren — APPROVED** [review]

File: app/views/comments/_form.html.erb (hunk 2 above)
Comment (jywarren) [reply app/views/comments/_form.html.erb]: I removed it!

**Comment by jywarren** [comment]
OK! I think we are good here for a first round! Thank you so much, @Cadreia for this big contribution, it's great! Let's merge and we can refine more in follow-up!

**Comment by jywarren** [comment]
Just confirming that images do exist on this, it's just that stable.publiclab.org doesn't have all the latest image assets as it's a test server!

**Pull request merged**